From 266f8d8e7f881c17c9390c4d3d95e2c7cf945791 Mon Sep 17 00:00:00 2001
Message-Id: <266f8d8e7f881c17c9390c4d3d95e2c7cf945791.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: vkorjani <vikas.korjani@intel.com>
Date: Fri, 19 Sep 2014 08:10:04 +0530
Subject: [PATCH 265/312] FOR_UPSTREAM [VPG]: drm/i915: Adding Power ON/OFF
 sequence and generic pmic operation

Adding power on/off sequence and generic pmic sequence for
Panels.

Issue: IRDA-821
Change-Id: Ib9b83fe8f5c876ad3a396a8b40b98ebf8dcaffae
Signed-off-by: vkorjani <vikas.korjani@intel.com>
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.h           |    2 ++
 drivers/gpu/drm/i915/intel_dsi_panel_vbt.c |   32 +++++++++++++++++++++++++---
 2 files changed, 31 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi.h b/drivers/gpu/drm/i915/intel_dsi.h
index 4d5dc9a..4e9f218 100644
--- a/drivers/gpu/drm/i915/intel_dsi.h
+++ b/drivers/gpu/drm/i915/intel_dsi.h
@@ -430,6 +430,8 @@ struct intel_dsi_dev_ops {
 	struct drm_display_mode *(*get_modes)(struct intel_dsi_device *dsi);
 
 	void (*destroy) (struct intel_dsi_device *dsi);
+	void (*power_on)(struct intel_dsi_device *dsi);
+	void (*power_off)(struct intel_dsi_device *dsi);
 };
 
 struct intel_dsi {
diff --git a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
index 45a0ea7..ac7c2c8 100644
--- a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
+++ b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
@@ -484,9 +484,12 @@ static const char * const seq_name[] = {
 	"MIPI_SEQ_DISPLAY_ON",
 	"MIPI_SEQ_DISPLAY_OFF",
 	"MIPI_SEQ_DEASSERT_RESET",
-	"MIPI_BACKLIGHT_ON",
-	"MIPI_BACKLIGHT_OFF",
-	"MIPI_TEAR_ON",
+	"MIPI_SEQ_BACKLIGHT_ON",
+	"MIPI_SEQ_BACKLIGHT_OFF",
+	"MIPI_SEQ_TEAR_ON",
+	"MIPI_SEQ_TEAR_OFF",
+	"MIPI_SEQ_PANEL_ON",
+	"MIPI_SEQ_PANEL_OFF"
 };
 
 static void generic_exec_sequence(struct intel_dsi *intel_dsi, char *sequence)
@@ -876,6 +879,26 @@ static struct drm_display_mode *generic_get_modes(struct intel_dsi_device *dsi)
 	return dev_priv->vbt.lfp_lvds_vbt_mode;
 }
 
+void generic_power_on(struct intel_dsi_device *dsi)
+{
+	struct intel_dsi *intel_dsi = container_of(dsi, struct intel_dsi, dev);
+	struct drm_device *dev = intel_dsi->base.base.dev;
+	struct drm_i915_private *dev_priv = dev->dev_private;
+
+	char *sequence = dev_priv->vbt.dsi.sequence[MIPI_POWER_ON];
+	generic_exec_sequence(intel_dsi, sequence);
+}
+
+void generic_power_off(struct intel_dsi_device *dsi)
+{
+	struct intel_dsi *intel_dsi = container_of(dsi, struct intel_dsi, dev);
+	struct drm_device *dev = intel_dsi->base.base.dev;
+	struct drm_i915_private *dev_priv = dev->dev_private;
+
+	char *sequence = dev_priv->vbt.dsi.sequence[MIPI_POWER_OFF];
+	generic_exec_sequence(intel_dsi, sequence);
+}
+
 static void generic_destroy(struct intel_dsi_device *dsi) { }
 
 /* Callbacks. We might not need them all. */
@@ -894,4 +917,7 @@ struct intel_dsi_dev_ops vbt_generic_dsi_display_ops = {
 	.get_hw_state = generic_get_hw_state,
 	.get_modes = generic_get_modes,
 	.destroy = generic_destroy,
+	.power_on = generic_power_on,
+	.power_off = generic_power_off,
+
 };
-- 
1.7.9.5

